nerfstudio/configs/config.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2022 The Nerfstudio Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Main config used for running an experiment"""
+
+from dataclasses import dataclass
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import yaml
+from rich.console import Console
+from typing_extensions import Literal
+
+from nerfstudio.configs.base_config import (
+    LoggingConfig,
+    MachineConfig,
+    PrintableConfig,
+    TrainerConfig,
+    ViewerConfig,
+)
+from nerfstudio.configs.config_utils import to_immutable_dict
+from nerfstudio.engine.optimizers import OptimizerConfig
+from nerfstudio.engine.schedulers import SchedulerConfig
+from nerfstudio.pipelines.base_pipeline import VanillaPipelineConfig
+
+CONSOLE = Console(width=120)
+
+
+@dataclass
+class Config(PrintableConfig):
```

Comment:
Maybe rename file and config `ExperimentConfig`

Reply:
that what I was thinking too..